### PR TITLE
65233 - Refacto mass entry

### PR DESCRIPTION
**Pull request by paa-axelor**

### CODE REVIEW

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java [hunk 1]
```diff
@@ -232,8 +232,187 @@ public Map<String, Map<String, Object>> getAccountOnChangeAttrsMap(
...
+    Map<String, Object> valuesMap = new HashMap<>();
+
+    if (moveLine.getInputAction() == MoveLineMassEntryRepository.MASS_ENTRY_INPUT_ACTION_MOVE) {
+      moveLine = new MoveLineMassEntry();
+      massEntryService.resetMoveLineMassEntry(moveLine);
+      moveLineMassEntryRecordService.setNextTemporaryMoveNumber(moveLine, move);
+      moveLine.setCounter(1);
+
+      valuesMap = this.setAllMoveLineValuesMap(moveLine);
+    }
```
Comment (aol-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java:340]: Same as before.
Also there should be only one method in the entire application that creates objects. Here it should call a method called `createMoveLineMassEntry` or something

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java [hunk 2]
```diff
@@ -54,10 +72,12 @@ public void resetDebit(MoveLineMassEntry moveLine) {
 
   @Override
   public void setMovePfpValidatorUser(MoveLineMassEntry moveLine, Company company) {
-    moveLine.setMovePfpValidatorUser(
-        Beans.get(MoveLineMassEntryService.class)
-            .getPfpValidatorUserForInTaxAccount(
-                moveLine.getAccount(), company, moveLine.getPartner()));
+    if (ObjectUtils.notEmpty(company)) {
+      moveLine.setMovePfpValidatorUser(
+          Beans.get(MoveLineMassEntryService.class)
```
Comment (aol-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java:77]: Please inject service

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java [hunk 3]
```diff
@@ -89,4 +109,119 @@ public void refreshAccountInformation(MoveLine moveLine, Move move) throws Axelo
...
+      moveLine.setVatSystemSelect(newMoveLine.getVatSystemSelect());
+      moveLine.setTaxLine(newMoveLine.getTaxLine());
+      moveLine.setAnalyticDistributionTemplate(newMoveLine.getAnalyticDistributionTemplate());
+      moveLine.setCurrencyCode(newMoveLine.getCurrencyCode());
+    }
+  }
+
+  @Override
+  public void setMovePaymentCondition(MoveLineMassEntry moveLine, int journalTechnicalTypeSelect) {
+    moveLine.setMovePaymentCondition(null);
```
Comment (aol-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java:158]: This should be in an `else` block

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java [hunk 4]
```diff
@@ -89,4 +109,119 @@ public void refreshAccountInformation(MoveLine moveLine, Move move) throws Axelo
...
+    }
+  }
+
+  @Override
+  public void setMovePaymentMode(MoveLineMassEntry moveLine, Integer technicalTypeSelect) {
+    switch (technicalTypeSelect) {
+      case 1:
+        moveLine.setMovePaymentMode(moveLine.getPartner().getOutPaymentMode());
+        break;
+      case 2:
```
Comment (aol-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java:170]: Please use static global constants

File: axelor-account/src/main/resources/views/MoveLineMassEntry.xml [hunk 5]
```diff
@@ -7,15 +7,15 @@
...
+      onChange="action-move-line-mass-entry-method-temporary-move-number-onchange"/>
     <field name="inputAction" selection="iaccount.move.line.mass.entry.input.action.select"
       onChange="action-move-line-mass-entry-method-input-action-onchange" readonly="true"
       required="true" width="50" selection-in="[1,2,3]"/>
     <field name="date" onChange="account-move-line-method-date-onchange">
       <hilite if="$contains(fieldsErrorList,'date:'+date.toString())" background="danger"/>
     </field>
     <field name="originDate"
-      onChange="action-move-line-mass-entry-method-set-currencyRate,action-move-line-mass-entry-attrs-set-cut-off-dates">
+      onChange="action-move-line-mass-entry-method-origin-date-onchange,action-move-line-mass-entry-attrs-set-cut-off-dates">
```
Comment (aol-axelor) [inline axelor-account/src/main/resources/views/MoveLineMassEntry.xml:18]: Please group those actions

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryAttrsServiceImpl.java [hunk 6]
```diff
@@ -151,10 +156,74 @@ public void addMovePaymentConditionRequired(
...
+  }
+
+  @Override
+  public void addPfpValidatorUserDomain(
+      Partner partner, Company company, Map<String, Map<String, Object>> attrsMap) {
+
+    this.addAttr(
+        "movePfpValidatorUser",
+        "domain",
+        Beans.get(InvoiceTermService.class).getPfpValidatorUserDomain(partner, company),
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryAttrsServiceImpl.java:183]: Resolved

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryAttrsServiceImpl.java [hunk 7]
```diff
@@ -151,10 +156,74 @@ public void addMovePaymentConditionRequired(
...
+    this.addAttr("currencyRate", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("currencyAmount", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("cutOffStartDate", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("cutOffEndDate", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("pfpValidatorUser", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("movePartnerBankDetails", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("vatSystemSelect", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("moveStatusSelect", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("account", "readonly", isCounterPartLine, attrsMap);
+    this.addAttr("analyticMoveLineList", "readonly", isCounterPartLine, attrsMap);
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryAttrsServiceImpl.java:207]: Resolved

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java (hunk 2 above)
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java:77]: Resolved

File: axelor-account/src/main/resources/views/MoveLineMassEntry.xml (hunk 5 above)
Comment (paa-axelor) [reply axelor-account/src/main/resources/views/MoveLineMassEntry.xml:18]: Resolved

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java (hunk 4 above)
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java:170]: Resolved

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java (hunk 3 above)
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryRecordServiceImpl.java:158]: Resolved

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java [hunk 8]
```diff
@@ -232,8 +232,187 @@ public Map<String, Map<String, Object>> getAccountOnChangeAttrsMap(
...
+        move.getMoveLineMassEntryList(), moveLine);
+
+    return this.setAllMoveLineValuesMap(moveLine);
+  }
+
+  @Override
+  public Map<String, Map<String, Object>> getTemporaryMoveNumberOnChangeAttrsMap(Move move) {
+    Map<String, Map<String, Object>> attrsMap = new HashMap<>();
+
+    if (move.getMoveLineMassEntryList() != null && move.getMoveLineMassEntryList().size() != 0) {
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java:369]: Resolved

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java [hunk 9]
```diff
@@ -232,8 +232,187 @@ public Map<String, Map<String, Object>> getAccountOnChangeAttrsMap(
...
+        move.setPartner(moveLine.getPartner());
+        moveLineMassEntryRecordService.setMovePaymentCondition(
+            moveLine, journalTechnicalTypeSelect);
+        moveLineMassEntryRecordService.loadAccountInformation(move, moveLine);
+        move.setPartner(null);
+        moveLineMassEntryRecordService.setVatSystemSelect(moveLine, move);
+      }
+      moveLineMassEntryRecordService.setMovePartnerBankDetails(moveLine);
+      moveLineMassEntryRecordService.setCurrencyCode(moveLine);
+    }
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java:295]: Resolved

**Pull request merged**